File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {

        int size = flowerbed.size();
        for( int i = 0; i < size; i++){
            bool left = (i == 0 || flowerbed[i-1] == 0);
            bool right = ( i == size -1 || flowerbed[i+1] == 0);
            if(flowerbed[i] == 0){
                if(left && right ){
                flowerbed[i] = 1;
                n--;

                if( n <= 0){
                    return true;
                }
            }

            }

        }



        return n <= 0;
        
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        f = [0] + flowerbed + [0]


        for i in range(1, len(f)-1):

            if n == 0:
                return True

            if f[i-1] == 0 and f[i] == 0 and f[i+1] == 0 and n > 0:
                f[i] = 1
                n -=1 

                if n == 0:
                    return True

        return n <= 0       